File: src/main/java/aubay/lu/projetrh/service/QcmService.java
package aubay.lu.projetrh.service;

import aubay.lu.projetrh.model.Qcm;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface QcmService {

    public List<Qcm> getAllQcm();

    public Optional<Qcm> getQcmById(UUID qcmId);

    public List<Qcm> getQcmByTitle(String title);

    public Qcm createQcm(Qcm qcm);

    public Qcm updateQcm(Qcm qcm);

    public String deleteQcm(UUID qcmId);

}

File: src/main/resources/projetrh/src/app/_services/_restricted/qcm.service.ts
import {Injectable} from "@angular/core";
import {Qcm} from "../../_models/qcm.model";
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {lastValueFrom, map, Observable, switchMap} from "rxjs";

@Injectable({
  providedIn:'root'
})

export class QcmService {

  qcmList!: Qcm[]

  constructor(private http: HttpClient) {}

  async getAllQcm(): Promise<Qcm[]>{
    let response = await lastValueFrom(this.http.get<Qcm[]>('http://localhost:8080/moderateur/qcm/all'))
    this.qcmList = response
    return this.qcmList
  }

  getQcmById(qcmId: any): Observable<Qcm>{
    return this.http.get<Qcm>(`http://localhost:8080/moderateur/qcm/id/${encodeURIComponent(qcmId)}`)
  }

  updateQcm(qcm: Qcm): Observable<Qcm>{
    return this.http.put<Qcm>('http://localhost:8080/moderateur/qcm/update', qcm)
  }

  createQcm(data: any, headers : {headers: HttpHeaders}): Observable<Qcm>{
    return this.http.post<Qcm>('http://localhost:8080/moderateur/qcm/create', data, headers)
  }

  deleteQcm(qcmId: any){
    this.http.delete(`http://localhost:8080/moderateur/qcm/delete/${encodeURIComponent(qcmId)}`).subscribe()
  }

}
